### ISSUE TITLE
Civilization: Beyond Earth

### ISSUE THREAD

**Issue by libcg** [libcg/grvk#41]
Meta-issue for CivBE.

- [ ] Working?
- [ ] ???

**Comment by niobium93** [comment]
Doesn't work. 
[grvk.log](https://github.com/libcg/grvk/files/7694265/grvk.log)
grvk_axl.log is empty.

**Comment by Cherser-s** [comment]
Looks like `CUBEMAP_ARRAY` dimension and `CULL_DISTANCE` imported register aren't handled.

**Comment by plasticbomb1986** [comment]
Hoi! Just made some testrun. Game starts up with DX11, crash with mantle with:

`Unhandled Exception
Code:  EXCEPTION_ACCESS_VIOLATION
Error writing address 0x6a317088

Call Stack
-----------------------------------------------------
(0x0000000000000000 : 0x6A317088) ??? ! ???  (???, line 0)
(0x0000000000000000 : 0x00A2882E) ??? ! ???  (???, line 0)
(0x0000000000000000 : 0x00A28EDD) ??? ! ???  (???, line 0)
(0x0000000000000000 : 0x00A185A6) ??? ! ???  (???, line 0)
(0x0000000000000000 : 0x00636B46) ??? ! ???  (???, line 0)
(0x0000000000000000 : 0x0051EF4E) ??? ! ???  (???, line 0)
(0x0000000000000000 : 0x0066AAB0) ??? ! ???  (???, line 0)
(0x0000000000000000 : 0x00A0C534) ??? ! ???  (???, line 0)
(0x0000000000000000 : 0x00A0C7EB) ??? ! ???  (???, line 0)
(0x0000000000000000 : 0x007D6D0F) ??? ! ???  (???, line 0)
(0x0000000000000000 : 0x00B01E54) ??? ! ???  (???, line 0)
(0x0000000000000000 : 0x7B62DCA0) ??? ! ???  (???, line 0)
(0x0000000000000000 : 0x7BC5D7F7) ??? ! ???  (???, line 0)
(0x0000000000000000 : 0x7BC5DEB0) ??? ! ???  (???, line 0)
`
[steam-65980.log](https://github.com/libcg/grvk/files/7740835/steam-65980.log)
[grvk.log](https://github.com/libcg/grvk/files/7740836/grvk.log)
[grvk_axl.log](https://github.com/libcg/grvk/files/7740837/grvk_axl.log)

**Comment by Cherser-s** [comment]
> I/00000118/grCreateDevice: 10005:0000 "llvmpipe (LLVM 13.0.0, 256 bits)" (Vulkan 1.2.199, driver 0.0.1)
why the hell it selects llvmpipe "physical device"?

**Comment by Cherser-s** [comment]
Also don't mind the shareable flag, but the game is actually crashing due to some flags in amd il translation being unimplemented.

**Comment by plasticbomb1986** [comment]
> > I/00000118/grCreateDevice: 10005:0000 "llvmpipe (LLVM 13.0.0, 256 bits)" (Vulkan 1.2.199, driver 0.0.1)
> 
> why the hell it selects llvmpipe "physical device"?

Not sure why it did (possible misconfiguration, i have two identical gpu (two vega64) in my system and the default vulkan behaviour for some weird reason mark the wrong GPU as primary (the actually primary is on pcie bus 0c and the secondary is in 0f, but vulkan just reverse that order and mark 0f as gpuid0 while the actual primary gets gpuid1), so i have to overwrite it with either mesa device select layer or with vkdevicechooser layer, and if i dont do it, it goes...anywhere.), but after i got rid of every other icd (what mesa-git def install) except the radv icds, it stopped doing that...



> Also don't mind the shareable flag, but the game is actually crashing due to some flags in amd il translation being unimplemented, which triggers the crash after attempting to create a shader module.

Sooo, its not supposed to work yet. Gotcha. Well, i will see notification about updates and when it will be about time i will be able to try to run the game with it.

**Comment by libcg** [comment]
@niobium93 @plasticbomb1986 can one of you run the game with [this build](https://github.com/libcg/grvk/suites/4708213536/artifacts/129087929) and `GRVK_DUMP_SHADERS=1`, zip up the files and email them to me? I need to see what the shader looks like to fix `unhandled import usage 7`. However the shaders are under copyright so they can't be shared publicly.

**Comment by plasticbomb1986** [comment]
In the afternoon i can, in about 9 hour.

**Comment by plasticbomb1986** [comment]
> @niobium93 @plasticbomb1986 can one of you run the game with [this build](https://github.com/libcg/grvk/suites/4708213536/artifacts/129087929) and `GRVK_DUMP_SHADERS=1`, zip up the files and email them to me? I need to see what the shader looks like to fix `unhandled import usage 7`. However the shaders are under copyright so they can't be shared publicly.


New log, theres something new, at least it seems new in it for me.
[grvk.log](https://github.com/libcg/grvk/files/7746890/grvk.log)

About to send shaders now.

**Comment by niobium93** [comment]
@libcg Sent the shaders, here's a new trace log:
[grvk.log](https://github.com/libcg/grvk/files/7747089/grvk.log)

**Comment by plasticbomb1986** [comment]
with trace here too
[steam-65980.log](https://github.com/libcg/grvk/files/7747419/steam-65980.log)
[grvk.log](https://github.com/libcg/grvk/files/7747421/grvk.log)
:

**Comment by libcg** [comment]
Thanks, I'll look into it soon. Focusing on DA:I right now.

**Comment by plasticbomb1986** [comment]
New set of logs:
[steam-65980.log](https://github.com/libcg/grvk/files/8309044/steam-65980.log)
[APP.log](https://github.com/libcg/grvk/files/8309045/APP.log)
[Renderer.log](https://github.com/libcg/grvk/files/8309046/Renderer.log)
[grvk.log](https://github.com/libcg/grvk/files/8309047/grvk.log)
[grvk_axl.log](https://github.com/libcg/grvk/files/8309048/grvk_axl.log)


I think its still the same story here.

**Comment by Cherser-s** [comment]
Welp it seems that `IL_BUF_DATA_FMT_8_8` isn't handled in AMD IL compiler.

**Comment by libcg** [comment]
Work in progress

![Screenshot from 2022-03-19 16-25-35](https://user-images.githubusercontent.com/490159/159141832-cdf72153-2162-4ef0-8f89-07b67c741a41.png)

**Comment by niobium93** [comment]
Just built latest master, still doesn't load the menu.
[grvk.log](https://github.com/libcg/grvk/files/8311058/grvk.log)
[wine.log](https://github.com/libcg/grvk/files/8311059/wine.log)
[APP.log](https://github.com/libcg/grvk/files/8311060/APP.log)

**Comment by libcg** [comment]
![Screenshot from 2022-05-01 12-35-49](https://user-images.githubusercontent.com/490159/167999528-f0e747d6-94ce-4000-91b9-9336f627dd6a.png)
it's going ingame with master + a few unmerged changes

**Comment by clarence-tbca** [comment]
I use the latest build "grvk-cf28ccc395b9b1a4099784df6160469ed2e3659e.zip" run this game
but it still didn't work
[grvk.log](https://github.com/libcg/grvk/files/8679732/grvk.log)

**Comment by libcg** [comment]
Yes, it needs some hacks to even reach the menu. I'll get these cleaned up and merged soon
